src/type/definition.ts

```diff
@@ -1534,13 +1534,7 @@ export class GraphQLEnumType /* <T> */ implements GraphQLSchemaElement {
     this.astNode = config.astNode;
     this.extensionASTNodes = config.extensionASTNodes ?? [];
 
-    this._values =
-      typeof config.values === 'function'
-        ? config.values
-        : Object.entries(config.values).map(
-            ([valueName, valueConfig]) =>
-              new GraphQLEnumValue(this, valueName, valueConfig),
-          );
+    this._values = defineEnumValues.bind(undefined, this, config.values);
```

Comment:
I'd be surprised if the `.bind()` method ends up faster than the simpler/shorter/clearer version:

```suggestion
    this._values = () => defineEnumValues(this, config.values);
```

Both require memory allocation, and traditionally `.bind()` has ended up slower than just using a new function, though the V8 team may have optimised that difference away by now!

Reply:
I could benchmark when I get a chance I think we are just using the convention from elsewhere in the file.

Reply:
I'd go with the simpler code if there's no perf difference. My guess is that `.bind()` was assumed to be more efficient, but I don't think it is.

Reply:
Hmmm. It seems in some scenarios to carry a small performance cost. I am wary of microbenchmarks but they are all we have at this point.

https://github.com/graphql/graphql-js/pull/4470

Reply:
Happy that the v8 team have optimised this! Always struck me as weird that a new function was cheaper in the past. Looks like the relevant V8 fix went out in Node 8.3… so a little while ago now! https://github.com/davidmarkclements/v8-perf?tab=readme-ov-file#partial-application-currying-and-binding